twilio/rest/__init__.py

```diff
@@ -116,11 +119,23 @@ def request(self, method, uri, params=None, data=None, headers=None, auth=None,
         if 'Accept' not in headers:
             headers['Accept'] = 'application/json'
 
-        if self.region:
-            head, tail = uri.split('.', 1)
+        if self.region or self.edge:
+            pieces = uri.split('.')
+            pieces[-2:] = ['.'.join(pieces[-2:])]
+
+            region = None
+            edge = None
+            if len(pieces) == 3:
+                region = pieces[1]
+            elif len(pieces) == 4:
```

Comment:
Comments on each of these checks would help to indicate an example syntax that we're looking for.